WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -693,6 +731,154 @@ final class ProductImageUploaderTests: XCTestCase {
+        if case let .failedUploadingImage(receivedAsset, receivedError) = receivedErrors.first?.error {
+            XCTAssertEqual(receivedAsset, asset)
+            XCTAssertEqual((receivedError as NSError).domain, error.domain)
+            XCTAssertEqual((receivedError as NSError).code, error.code)
+        } else {
+            XCTFail("Expected failedUploadingImage error")
+        }
+    }
+
+    func test_activeUploads_are_tracked_through_storage_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+
+        _ = CurrentValueSubject<[ProductImageUploaderKey], Never>([])
+        let mockImageUploader = MockProductImageUploader()
+        mockImageUploader.activeUploadsKeys = []
+
+        var activeUploads: [ProductImageUploaderKey] = []
+        activeUploadsSubscription = mockImageUploader.activeUploads.sink { keys in
+            activeUploads = keys
+        }
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        // When - simulate an uploading status
+        mockImageUploader.activeUploadsKeys = [key]
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 1)
+        XCTAssertEqual(activeUploads.first?.siteID, key.siteID)
+        XCTAssertEqual(activeUploads.first?.productOrVariationID, key.productOrVariationID)
+        XCTAssertEqual(activeUploads.first?.isLocalID, key.isLocalID)
+
+        // When - simulate completion of upload
+        mockImageUploader.activeUploadsKeys = []
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 0)
+    }
+
+    func test_background_upload_notice_is_sent_when_there_are_active_uploads_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+
+        let mockImageUploader = MockProductImageUploader()
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        let noticePresenter = MockNoticePresenter()
+
+        // When - No active uploads
+        mockImageUploader.activeUploadsKeys = []
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+
+        // Reset flag for next check
+        mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled = false
+
+        // When - With active uploads
+        mockImageUploader.activeUploadsKeys = [key]
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+    }
```

Comment:
Same as the other comments about tests. This test should be testing `ProductImageUploader`. Instead, it creates an instance of `MockProductImageUploader` and tests the mock.